update docs for HostNavigator related changes (#827)
freeletics/khonshu: docs/navigation/activities.md

# Navigating to Activities and other apps

It is also possible navigate to an `Activity` both inside and outside of the app. Similar to other
screens it is required to define a route for Activities. Instead of `NavRoute`
the route class needs to extend either `InternalActivtyRoute` for `Activity` instances in the
current app or `ExternalActivityRoute` if it is part of a different app.

## Setup

Navigating to an `Activity` is not directly possible with `HostNavigator`. Instead an
`ActivityNavigator` is needed. This can just be a simple class that extends `ActivityNavigator`
or `DestinationNavigator`. The latter combines the functionality of `HostNavigator` and
`ActivityNavigator` so that one class can be used for all navigation related actions of a
screen.

From the `Composable` it is then required to call `NavigationSetup(navigator)` which

> will do the required setup to make the `Activity` related navigation actions work.

Reviewer: Since this is done by codegen, it is probably not needed to be mentioned here, unless codegen isn't used. What do you think?

Author: I mostly mentioned it to keep the navigation docs codegen free and describe how it would work when doing it manually. Similar to how it's also talking about how to define the destinations

It's recommended to make `ActivityNavigator`/`DestinationNavigator` classes specific to
one screen where they are needed instead of having a global instance.

## Internal Activities

This is example shows the route for a `SettingsActivity`:
```kotlin
@Parcelize
data class SettingsActivityRoute(
    val id: String,
) : InternalActivityRoute() {
    override fun buildIntent() = Intent("com.example.SETTINGS")
}
```

An instance of the route will automatically be added to the `Intent` which is why it's not needed to
manually put the `id` into it. `SettingsActivity` can then obtain an instance of the route that was used
to navigate to it by calling `Activity.getRoute()` or `Activity.requireRoute()` and access the
argument through it.

> To avoid having to reference a specific `Activity` `class` it's possible to use customt `Intent` actions.

Reviewer: typo

```suggestion
To avoid having to reference a specific `Activity` `class` it's possible to use custom `Intent` actions.
```

        })
}

// route with a data uri extra
@Parcelize
class BrowserRoute(
    private val uri: Uri,
) : ExternalActivityRoute {
    // the returned Intent is filled into the Intent of the destination by calling
    // destinationIntent.fillIn(fillInIntent())
    override fun fillInIntent() = Intent(Intent.ACTION_VIEW).setData(uri)
}
```

## Activity results

External SDKs and the framework often provide an `Activity` that is supposed to be used with
`startActivityForResult`. AndroidX already introduced `ActivityResultContract` to simplify handling

> this and `ActivityNavigator` uses them to also enable starting them from outside the UI layer

Reviewer: rephrased since it's not obvious what `them` is

```suggestion
this and `ActivityNavigator` uses it to also enable starting Activities from outside the UI layer
```

and receiving results there.

To use the API `registerForActivityResult` needs to be called with an instance of the wanted
`ActivityResultContract`. This needs to happen before `NavigationSetup` is called for the navigator,

> so it is recommended to do this during the construction of the navigator. The method returns an

Reviewer: > This needs to happen before `NavigationSetup` is called for the navigator,
so it is recommended to do this during the construction of the navigator. 

Isn't it the only way now? `registerForActivityResult` has to be done either in constructor or as `val` property of navigator. Anything else would be late and cause runtime crash.

Author: I think there are some ways where you could do the registration between constructor and setup but let's change it:
```suggestion
so it needs to be called during the construction of the navigator. The method returns an
```
